constants.lua

```diff
@@ -44,49 +44,29 @@ local CONSTANTS = {
     -- we do expensive upgrades first (instead of just letting the road always expand).
     -- Sepcial buildings (like the treasury) are an exception that should ideally come first.
     CONSTRUCTION_MATERIALS = {
-        specialBuildings = {{
-            name = "stone-brick",
-            required = 1,
-        }, {
-            name = "iron-plate",
-            required = 1,
-        }},
-        highrise = {{
-            name = "concrete",
-            required = 50,
-        }, {
-            name = "steel-plate",
-            required = 25,
-        }, {
-            name = "small-lamp",
-            required = 5,
-        }, {
-            name = "pump",
-            required = 2,
-        }, {
-            name = "pipe",
-            required = 10,
-        }},
-        residential = {{
-            name = "stone-brick",
-            required = 30,
-        }, {
-            name = "iron-plate",
-            required = 20,
-        }, {
-            name = "steel-plate",
-            required = 10,
-        }, {
-            name = "small-lamp",
-            required = 2,
-        }},
-        simple = {{
-            name = "stone-brick",
-            required = 10,
-        }, {
-            name = "iron-plate",
-            required = 5,
-        }},
+        ["tycoon-treasury"] = {
+            ["stone-brick"] = 1,
+            ["iron-plate"] = 1,
+        },
+        highrise = {
+            ["concrete"] = 50,
+            ["steel-plate"] = 25,
+            ["small-lamp"] = 5,
+            ["pump"] = 2,
+            ["pipe"] = 10,
+        },
+        residential = {
+            ["stone-brick"] = 30,
+            ["iron-plate"] = 20,
+            ["steel-plate"] = 10,
+            ["small-lamp"] = 2,
+        },
+        simple = {
+            ["stone-brick"] = 10,
+            ["iron-plate"] = 5,
+        },
+        garden = {
+        },
```

Comment:
@bahrmichael , as this now looks much easier to read - please recheck this table(dict) as i don't know global structure and initial design about all of this. it might be missing something

note: **all valid building must be listed here** from now on, otherwise `assert()` is raised from `Consumption:: consumeRequired()` to avoid errors before releasing
free construction such as current `garden` must be listed as `{}` empty table
oh, and `"specialBuildings"` was not valid entity name, so i reworked it to (almost full) names

Reply:
> note: all valid building must be listed here from now on, otherwise assert() is raised from Consumption:: consumeRequired() to avoid errors before releasing

Nice!